Implementing support for archive URL and date

BX326 opened the pull request:
See #9. Still needs a little work, notably tests and removal of at least one ``unwrap()``.

BX326 committed:
Partial support for archived URL and date

An archived URL and date can now be fetched from the Wayback Machine API in a synchronous manner. The ArchiveOptions aren't being taken into consideration yet, but that's an easy adjustment.

BX326 committed:
WikiCitation uses archived URL and date

Still need to act according to the user-specified ArchiveOptions. We should also write some unit tests for this.

philipwastakenwastaken commented:
I think this is OK to merge for now. I think the main issue is that calling the `archive.org` endpoint should be done asynchronously, as it can take quite a long time. This can done in a separate PR, and I'll create an issue for this after merging.

Commit:
Merge branch 'dev' into iss9

Commit:
Use curl.rs for retrieving archival requests and add a test case

Our preferred way to perform web requests is using
curl and our own curl.rs module. This was included in a newer PR,
before it could have been included here. A single test case
for a known archived article has been created.

philipwastakenwastaken commented:
I've unified the wayback machine API calls such that they use `curl.rs` and deleted the old `utils.rs` module which provided the same functionality. I've also added a single test case to confirm that we can actually return archived links.
